app/src/main/java/space/taran/arknavigator/mvp/model/UserPreferences.kt

```diff
@@ -0,0 +1,42 @@
+package space.taran.arknavigator.mvp.model
+
+import android.content.Context
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.intPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.flow.first
+import space.taran.arknavigator.utils.Sorting
+import space.taran.arknavigator.utils.extensions.convertToSorting
+import javax.inject.Inject
+
+class UserPreferences @Inject constructor(val context: Context) {
+    private val SHARED_PREFERENCES_KEY = "user_preferences"
+
+    private val Context.preferencesDatastore by preferencesDataStore((SHARED_PREFERENCES_KEY))
+
+    private val dataStore = context.preferencesDatastore
+
+    suspend fun setUserSorting(sorting: Sorting) {
```

Comment:
Just `setSorting` and `setSortingAscending` should be enough. There is already `User` word in the name of the class, so no confusion possible.